Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -35,76 +35,80 @@ extension PackageGraph {
     ) throws -> PackageGraph {
 
         // Create a map of the manifests, keyed by their identity.
-        //
-        // FIXME: For now, we have to compute the identity of dependencies from
-        // the URL but that shouldn't be needed after <rdar://problem/33693433>
-        // Ensure that identity and package name are the same once we have an
-        // API to specify identity in the manifest file
-        let manifestMapSequence = (root.manifests + externalManifests).map({ (identityResolver.resolveIdentity(for: $0.packageLocation), $0) })
-        let manifestMap = Dictionary(uniqueKeysWithValues: manifestMapSequence)
+        let rootManifestsMap = root.packages.mapValues { $0.manifest }
+        let externalManifestsMap = externalManifests.map{ (identityResolver.resolveIdentity(for: $0.packageLocation), $0) }
+        let manifestMap = rootManifestsMap.merging(externalManifestsMap, uniquingKeysWith: { lhs, rhs in
+            return rhs // 👀 this was not possible before (the dictionary creation would crash), is preferring external correct?
```

Comment:
It seems to me that preferring roots over externals would be preferable, since that would preserve the current overriding semantics.

Reply:
changes to `lhs`